python/ray/dag/dag_operation_future.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Generic, TypeVar
+from ray.util.annotations import DeveloperAPI
+
+
+if TYPE_CHECKING:
+    import torch
+    import cupy as cp
+
+T = TypeVar("T")
+
+
+@DeveloperAPI
+class DAGOperationFuture(ABC, Generic[T]):
+    """
+    A future representing the result of a DAG operation.
```

Comment:
Add a comment explaining that this abstraction is internal to each actor, not exposed to the DAG caller.